test/Npgsql.Tests/MultipleHostsTests.cs

```diff
@@ -963,6 +963,117 @@ public void DataSource_with_single_host_is_not_supported()
             .With.Message.EqualTo(NpgsqlStrings.MultipleHostsMustBeSpecified));
     }
 
+    [Test, IssueLink("https://github.com/npgsql/npgsql/issues/4181")]
+    [Explicit("Fails until #4181 is fixed.")]
+    public async Task LoadBalancing_is_fair_if_first_host_is_down([Values]TargetSessionAttributes targetSessionAttributes)
+    {
+        using var socket = new Socket(AddressFamily.InterNetwork, SocketType.Stream, ProtocolType.Tcp);
+        var endpoint = new IPEndPoint(IPAddress.Loopback, 0);
+        socket.Bind(endpoint);
+        var localEndPoint = (IPEndPoint)socket.LocalEndPoint!;
+        // Note that we Bind (to reserve the port), but do not Listen - connection attempts will fail.
```

Comment:
You can also start `PgPostmasterMock` with an error code, like `PostgresErrorCodes.CannotConnectNow` (test `All_hosts_are_unavailable` as an example).